core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryParser.java

```diff
@@ -50,31 +45,41 @@ public MultiMatchQueryParser() {
     }
 
     @Override
-    public Query parse(QueryShardContext context) throws IOException, QueryParsingException {
-        QueryParseContext parseContext = context.parseContext();
+    public MultiMatchQueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
 
         Object value = null;
-        float boost = AbstractQueryBuilder.DEFAULT_BOOST;
-        Float tieBreaker = null;
+        Map<String, Float> fieldsBoosts = Maps.newHashMap();
         MultiMatchQueryBuilder.Type type = null;
-        MultiMatchQuery multiMatchQuery = new MultiMatchQuery(context);
+        String analyzer = null;
+        Integer slop = null;
+        Fuzziness fuzziness = null;
+        Integer prefixLength = null;
+        Integer maxExpansions = null;
+        Operator operator = null;
         String minimumShouldMatch = null;
-        Map<String, Float> fieldNameWithBoosts = Maps.newHashMap();
+        String fuzzyRewrite = null;
+        Boolean useDisMax = null;
+        Float tieBreaker = null;
+        Float cutoffFrequency = null;
+        Boolean lenient = null;
+        MatchQuery.ZeroTermsQuery zeroTermsQuery = null;
```

Comment:
We can have ZeroTermsQuery.NONE here as a default, thats what MatchQuery uses. Then some null-checks can be removed later. I added a default constant to MatchQuery here: https://github.com/elastic/elasticsearch/pull/13402/files#diff-943345b14010c833dc54a52b10a28489R128